lib/sqlalchemy/dialects/oracle/base.py

```diff
@@ -850,9 +1045,91 @@
     "BINARY_DOUBLE": BINARY_DOUBLE,
     "BINARY_FLOAT": BINARY_FLOAT,
     "ROWID": ROWID,
+    "VECTOR": VECTOR,
 }
 
 
+class VectorIndexType(Enum):
+    HNSW = "HNSW"
+    IVF = "IVF"
+
+
+class VectorDistanceType(Enum):
+    EUCLIDEAN = "EUCLIDEAN"
+    DOT = "DOT"
+    COSINE = "COSINE"
+    MANHATTAN = "MANHATTAN"
+
+
+@dataclass
+class VectorIndexConfig:
```

Comment:
this needs a docstring also, but it also has to list out these params.  since there is no `__init__` method, let's try documenting the params here.  Additionally, as they seem to be specific to various `VectorIndexType` combinations, that should be here as well:

```py
@dataclass
class VectorIndexConfig:
     """Defines the configuration for < accurate terminology here >

      :param index_type: the <XYZ> - applies to :attr:`.VectorIndexType.HNSW`, <others>
      :param distance: <QPR> - applies to :attr:`.VectorIndexType.FOO`, :attr:`.VectorIndexType.BAR`
      # ... etc.

```